linker/shared/src/main/scala/org/scalajs/linker/standard/LinkTimeProperties.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.standard
+
+import org.scalajs.ir.{Types => jstpe, Trees => js}
+import org.scalajs.ir.ScalaJSVersions
+import org.scalajs.ir.Position.NoPosition
+import org.scalajs.linker.interface.{Semantics, ESFeatures}
+
+final class LinkTimeProperties private[standard] (
+  semantics: Semantics,
+  esFeatures: ESFeatures,
+  targetIsWebAssembly: Boolean
+) {
+  import LinkTimeValue._
+
+  private val linkTimeProperties: Map[String, LinkTimeValue] = Map(
+    "core/esVersion" ->
+      LinkTimeInt(esFeatures.esVersion.edition),
+    "core/assumingES6" ->
```

Comment:
Let's take this opportunity to fix the mistakes of our past:
```suggestion
    "core/useECMAScript2015Semantics" ->
```